uvtools/dspec.py

```diff
@@ -216,12 +271,21 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
         d_mdl = np.fft.fft(_d_cl)
 
     # get residual in data space
-    d_res = (data - d_mdl) * ~np.isclose(wgts * win, 0.0)
+    if not linear:
+        d_res = (data - d_mdl) * ~np.isclose(wgts * win, 0.0)
+    else:
+        if clean2d:
+            d_res = np.fft.fft2(_d_res)
+        else:
+            d_res = np.fft.fft(_d_res, axis = 0)
+
 
     return d_mdl, d_res, info
 
-def linear_delay_filter(data, wgts, df, filter_centers, filter_widths, filter_factors, cache = {}):
+def linear_delay_filter(data, wgts, df, filter_centers, filter_widths, filter_factors, cache = {},
+                        clean_dimensions = [False, True]):
```

Comment:
I thought we were avoiding CLEAN terminology? perhaps `filter_dimensions`?

Reply:
Good point. I'll go ahead and change this.